waiter/src/waiter/state.clj

```diff
@@ -82,27 +115,44 @@
     (->> instance-id->state
          (filter (fn [[_ state]] (killable? state)))
          (filter (fn [[instance-id _]] (acceptable-instance-id? instance-id)))
+         ; if there are expired instances, only choose healthy instances that are blacklisted or expired
```

Comment:
Would this preference (and the preference below it related to starting instances), be better expressed inside our comparator?

Reply:
I don't think this would work, because you could only have the "replacement" instance in the list at this point (assuming the expired one was filtered out by `killable?`)